src/irmin/store.ml

```diff
@@ -849,27 +857,67 @@ module Make (B : Backend.S) = struct
         if Tree.equal test v then set_tree_once root key ~new_tree ~current_tree
         else err_test current_tree
 
-  let test_and_set_tree ?(retries = 13) ?allow_empty ?parents ~info t k ~test
-      ~set =
+  let test_and_set_tree_return_commit ?(retries = 13) ?allow_empty ?parents
+      ~info t k ~test ~set =
     [%log.debug "test-and-set %a" pp_path k];
     retry ~retries @@ fun () ->
     update t k ?allow_empty ?parents ~info (test_and_set_tree_once ~test)
     @@ fun _tree -> Lwt.return set
 
+  let test_and_set_tree_return_commit ?(retries = 13) ?allow_empty ?parents
+      ~info t k ~test ~set =
+    [%log.debug "test-and-set %a" pp_path k];
+    test_and_set_tree_return_commit ~retries ?allow_empty ?parents ~info t k
+      ~test ~set
+
+  let test_and_set_tree ?(retries = 13) ?allow_empty ?parents ~info t k ~test
+      ~set =
+    [%log.debug "test-and-set %a" pp_path k];
+    ignore_commit
+    @@ test_and_set_tree_return_commit ~retries ?allow_empty ?parents ~info t k
+         ~test ~set
+
   let test_and_set_tree_exn ?retries ?allow_empty ?parents ~info t k ~test ~set
       =
     test_and_set_tree ?retries ?allow_empty ?parents ~info t k ~test ~set
     >>= fail "test_and_set_tree_exn"
 
-  let test_and_set ?retries ?allow_empty ?parents ~info t k ~test ~set =
+  let test_and_set_return_commit ?retries ?allow_empty ?parents ~info t k ~test
```

Comment:
Same comment about getting rid of the helper functions by rearranging the function declarations.